https://bugs.gentoo.org/917760
We can't build program without library, utils dependency is just in case
diff -ru a/Makefile b/Makefile
--- a/Makefile	2025-01-09 11:48:10.117085442 +0400
+++ b/Makefile	2025-01-09 11:48:27.041979565 +0400
@@ -19,10 +19,10 @@
 lib:
 	$(MAKE) -C libflake all
 
-progs:
+progs: lib
 	$(MAKE) -C flake all
 
-utils:
+utils: progs lib
 	$(MAKE) -C util all
 
 .PHONY: install
https://bugs.gentoo.org/912148
https://bugs.gentoo.org/880819
False dependency, if we put flake in build graph, make uses default build
rules, doesn't include library and failure to link occurs.
diff -ru a/flake/Makefile b/flake/Makefile
--- a/flake/Makefile	2025-01-09 11:57:17.279914146 +0400
+++ b/flake/Makefile	2025-01-09 11:59:25.559155784 +0400
@@ -21,6 +21,6 @@
 
-all: $(PROGS_G) $(PROGS)
+all: $(PROGS_G)
 
 flake_g$(EXESUF): flake.o wav.o $(DEP_LIBS)
 	$(CC) $(FLAKE_LIBDIRS) $(LDFLAGS) -o $@ flake.o wav.o $(FLAKE_LIBS) $(EXTRALIBS)
 	cp -p flake_g$(EXESUF) flake$(EXESUF)
